a3fe/run/simulation.py

```diff
@@ -213,26 +211,13 @@ def _validate_input(self) -> None:
             if not _os.path.isfile(_os.path.join(self.input_dir, file)):
                 raise FileNotFoundError("Required input file " + file + " not found.")
 
-    def _add_attributes_from_simfile(self) -> None:
-        """
-        Read the SOMD simulation option file and
-        add useful attributes to the Simulation object.
-        All times in ns.
-        """
-
-        timestep = None  # ns
-        nrg_freq = None  # number of timesteps between energy calculations
-        timestep = float(
-            _read_simfile_option(self.simfile_path, "timestep").split()[0]
-        )  # Need to remove femtoseconds from the end
-        nrg_freq = float(_read_simfile_option(self.simfile_path, "energy frequency"))
-
-        self.timestep = timestep / 1_000_000  # fs to ns
-        self.nrg_freq = nrg_freq
-
     def _select_input_files(self) -> None:
         """Select the correct rst7 and, if supplied, restraints,
         according to the run number."""
+
+        # First ensure the most up-to-date SOMD configuration is loaded
+        self.engine_config.get_somd_config(self.input_dir)
```

Comment:
Let's remove  get_somd_config and move the logic here - slightly personal preference, but I feel like the logic belongs here. Thanks